src/js/contentscripts/dnt.js

```diff
@@ -26,22 +26,28 @@ function getPageScript() {
       OBJECT.defineProperty(OBJECT.getPrototypeOf(NAVIGATOR), "doNotTrack", {
         get: function doNotTrack() {
           return "1";
-        }
+        },
+        configurable: true,
+        enumerable: true
       });
     }
 
-    if (!NAVIGATOR.globalPrivacyControl) {
+    if (!globalThis.navigator.globalPrivacyControl) {
       try {
-        OBJECT.defineProperty(NAVIGATOR, "globalPrivacyControl", {
-          value: true
+        OBJECT.defineProperty(NAVIGATOR.prototype, "globalPrivacyControl", {
+          get: function () {
+            return true;
+          },
+          configurable: true,
+          enumerable: true
         });
       } catch (e) {
         console.error("Privacy Badger failed to set navigator.globalPrivacyControl, probably because another extension set it in an incompatible way first.");
       }
     }
 
   // save locally to keep from getting overwritten by site code
-  } + "(window.navigator, Object));";
+  } + "(self.Navigator, Object));";
```

Comment:
This should stay as `window.navigator`.

- `window` vs `self`: we use `window` everywhere now and `self` nowhere. Unless there is a good reason to switch, we shouldn't.
- `Navigator` is not the same thing as `navigator`

Reply:
> window vs self: we use window everywhere now and self nowhere. Unless there is a good reason to switch, we shouldn't.

`self` works in every environment that supports window, but also supports worker threads. I've gone ahead and reverted this change.

> `Navigator` is not the same thing as `navigator`

Indeed. The whole reason for this PR is to fix your patcher that is trying to directly extend window.navigator instead of Navigator. The standard specifies `globalPrivacyControl` on the `Navigator` interface, not only the specific `Navigator` instance.

Transcend Consent Manager is internally resolving `navigator.globalPrivacyControl` by effectively calling something similar to `Object.getOwnPropertyDescriptor(Navigator.prototype, 'globalPrivacyControl')?.get?.call(navigator)`. Until this PR is merged, Privacy Badger does not correctly indicate GPC preferences to Transcend Consent Manager.